### PR TITLE
fix: Fix thread issue with error  strategy for diff parsers.

### PR DESCRIPTION
**Pull request by ap891843**
Signed-off-by: ap891843 <aman.prashant@broadcom.com>

### CODE REVIEW

**Review by temanbrcom — APPROVED** [review]
🍻

**Comment by temanbrcom** [comment]
I will check the memory leak. 
Please, just change the commit message to make it start with an imperative verb.

**Review by temanbrcom — APPROVED** [review]
Please, remove also the binding in the EngineModule class and the other guice annotation from the CobolErrorStrategy

**Review by temanbrcom — CHANGES_REQUESTED** [review]
Please, remove also the binding in the EngineModule class and the other guice annotation from the CobolErrorStrategy

**Review by temanbrcom — APPROVED** [review]
🌳

File: server/src/main/java/org/eclipse/lsp/cobol/core/strategy/ErrorMessageUtil.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  static final String REPORT_UNWANTED_TOKEN = "ErrorStrategy.reportUnwantedToken";
+  static final String END_OF_FILE_MESSAGE = "ErrorStrategy.endOfFile";
+  static final String REPORT_INPUT_MISMATCH = "ErrorStrategy.reportInputMismatch";
+  private static final String MSG_DELIMITER = ", ";
+  private static final String MSG_PREFIX = "{";
+  private static final String MSG_SUFFIX = "}";
+  private static final Map<Class<? extends Parser>, Set<String>> IDENTIFIER_TOKENS =
+      IdentifierReplacing.retrieveTokenToRemove();
+
+  ErrorMessageUtil(MessageService messageService) {
```
Comment (grianbrcom) [inline server/src/main/java/org/eclipse/lsp/cobol/core/strategy/ErrorMessageUtil.java:46]: Sorry, but I disagree. A utility class can't be instantiated. Can we rename it to ErrorMessageHelper or Service, or whatever?

**Review by grianbrcom — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/eclipse/lsp/cobol/core/strategy/ErrorMessageUtil.java (hunk 1 above)
Comment (ap891843) [reply server/src/main/java/org/eclipse/lsp/cobol/core/strategy/ErrorMessageUtil.java]: Yes, I agree. Updated

**Review by grianbrcom — APPROVED** [review]
:safety_pin:

File: server/src/main/java/org/eclipse/lsp/cobol/core/strategy/ErrorMessageUtil.java (hunk 1 above)
Comment (grianbrcom) [reply server/src/main/java/org/eclipse/lsp/cobol/core/strategy/ErrorMessageUtil.java]: Thanks)

**Pull request merged**